src/datadog/w3c_propagation.cpp

```diff
@@ -62,18 +63,23 @@ Optional<std::string> extract_traceparent(ExtractedData& result,
   }
 
   assert(match.ready());
-  assert(match.size() == 4 + 1);
+  assert(match.size() == 5 + 1);
 
   const auto to_string_view = [](const auto& submatch) {
     assert(submatch.first <= submatch.second);
     return StringView{submatch.first,
                       std::size_t(submatch.second - submatch.first)};
   };
 
-  if (to_string_view(match[1]) == "ff") {
+  const auto version = to_string_view(match[1]);
```

Comment:
The only value that is not supported is `ff`.

I added the case with `00` because only with `00` do we reject trailing hyphen-separated fields. If it were, say, `02`, we tolerate them. This is per the shared tests.